Standards/scs-XXXX-v1-security-of-iaas-service-software.md

```diff
@@ -0,0 +1,108 @@
+Also CSPs that want to become SCS-compliant may not have the burden to upgrade their deployments, but can test before an upgrade, where they need to put in additional work to become SCS-compliant.
+
+One problem is, that there might be new features implemented in the newest versions of the software, which are desired by other standards to be SCS-compliant.
+In that case allowing all maintained versions would lead to a two-year timespan customers would need to wait for before such a feature becomes available in all SCS-compliant deployments.
+In case of security relevant features this is not advisable.
+
+#### Standards implicitly define the minimum versions of Software
+
+Instead of requiring a defined minimum software version, it could be derived from the standards.
+Because: Whenever there is a new wanted behavior a standard should be created and a resonable timeframe given to CSPs to adopt a software version that can fulfill the new standard.
+Through the combination of all standards that are in place, the minimum version for the IaaS service software is implicitly given.
+
+This would avoid to have conflicting versions of software in terms of feature parity, while also allowing older software.
+Using this approach requires an additional advise to CSPs to update or implement patches for security issues.
+
+#### Advise CSPs to integrate software updates
+
+As long as maintained versions of software are used, updates with security patches are available and only need to be integrated.
+This can and should be done in a reasonable short timeframe.
+
+But CSPs may even use releases of IaaS software, that are either not maintained anymore by an open source community or may be even closed source implementations of the mandatory IaaS APIs.
+Allowing older versions or closed source software would only be acceptable, when CSPs assure (e.g. in documentation), that they themself will patch the software within their deployments.
+Security bug fixes must be implemented and proof of the fix then provided.
+Only under these circumstances deployments with older or alternative IaaS Layer software may be handled as compliant.
+
+This option could be taken for granted, but to actually advise using it may encourage CSPs to take a closer look on their life-cycle management and security risk handling.
+And CSPs using OpenStack could even be encouraged to upgrade their deployments.
+
+## Standard for a minimum IaaS Layer Software version
+
+If a deployment is affected by a security issue and a maintained[^1] version of OpenStack is used as implementation for IaaS Layer software, security patches noted in OSSNs and OSSAs MUST be integrated within a reasonable timeframe.
+Otherwise the CSP MUST implement security bug fixes themself within a reasonable timeframe, when the deplyoment is affected by a security issue.
+
+In both cases proof of the update MUST be send to the OSBA, so that the compliance will not be revoked.
```

Comment:
I'm afraid, this will result in a lot of paper work. Isn't there any possibility to get software version, automatically?

Reply:
That really is a problem:
We want to create standards that should be followed. Therefore we need tests.
But future CVEs or OSSNs cannot be known now and may be not easily testable. Should we consider writing a test for each of these vulnerabilities? This might be several tests a year to write, if a test from the outside is possible at all. And those tests might be considered as attacks (I mean they somehow are ;) ) and might trigger responses on the side of CSPs.

So a CSP giving notice about the fixing of a vulnerability, which already implicates, that we need to trust the CSP, could be an option. Maybe - if we already need to trust the CSPs - we may reduce the paper work:
MArkus suggested that there should be some kind of form file, that could be provided. We could do something like:

```
How did you deal with the vulnerability on your deployment XYZ:
[ ] Deployment was NOT affected, because ....... (e.g. we use another backend)
[ ] Functionality was disabled in the deployment.
[ ] Deployment was updated with a fix for the vulnerability, that .... (e.g. was provided be upstream openstack)
```

And only require short descriptions like above.
This may even be easier to evaluate in the compliance checker?